src/Domain/Item/ItemService.ts

```diff
@@ -49,8 +55,138 @@ export class ItemService implements ItemServiceInterface {
+    const itemsKeys = await this.itemRepository.findAll({
+      userUuid,
+      contentType: Item.CONTENT_TYPE_ITEMS_KEYS,
+      sortBy: 'updated_at_timestamp',
+      sortOrder: 'DESC'
+    })
+
+    const retrievedItemsIds: Array<string> = retrievedItems.map((item: Item) => item.uuid)
+
+    itemsKeys.forEach((itemKey: Item) => {
+      if(retrievedItemsIds.indexOf(itemKey.uuid) === -1) {
+        retrievedItems.unshift(itemKey)
+      }
+    })
+
+    return retrievedItems
+  }
+
+  private calculateSyncToken(lastUpdatedTimestamp: number, savedItems: Array<Item>): string {
+    if (savedItems.length) {
+      const sortedItems = savedItems.sort((itemA: Item, itemB: Item) => itemA.updatedAt > itemB.updatedAt ? 1 : -1)
+      lastUpdatedTimestamp = sortedItems[sortedItems.length - 1].updatedAt
+    }
+
+    const lastUpdatedTimestampWithMicrosecondPreventingSyncDoubles = lastUpdatedTimestamp + 1
+
+    return Buffer.from(
+      `${this.SYNC_TOKEN_VERSION}:${lastUpdatedTimestampWithMicrosecondPreventingSyncDoubles / 1000000}`,
+      'utf-8'
+    ).toString('base64')
+  }
+
+  private async updateExistingItem(existingItem: Item, itemHash: ItemHash, userAgent: string): Promise<Item> {
+    existingItem.content = itemHash.content
+    existingItem.contentType = itemHash.content_type
+    if (itemHash.deleted !== undefined) {
+      existingItem.deleted = itemHash.deleted
+    }
+    if (itemHash.duplicate_of) {
+      existingItem.duplicateOf = itemHash.duplicate_of
+    }
+    existingItem.encItemKey = itemHash.enc_item_key
+    existingItem.itemsKeyId = itemHash.items_key_id
+    existingItem.lastUserAgent = userAgent
+
+    if (itemHash.deleted === true) {
+      existingItem.deleted = true
+      existingItem.content = null
+      existingItem.encItemKey = null
+      existingItem.authHash = null
+    }
+
+    existingItem.createdAt = dayjs.utc(itemHash.created_at).valueOf() * 1000
+    existingItem.updatedAt = this.timer.getTimestampInMicroseconds()
+
+    return this.itemRepository.save(existingItem)
+  }
+
+  private async saveNewItem(itemHash: ItemHash, userAgent: string): Promise<Item> {
+    const newItem = new Item()
+    newItem.uuid = itemHash.uuid
+    newItem.content = itemHash.content
+    newItem.contentType = itemHash.content_type
+    newItem.encItemKey = itemHash.enc_item_key
+    newItem.itemsKeyId = itemHash.items_key_id
+    newItem.lastUserAgent = userAgent
+    const now = this.timer.getTimestampInMicroseconds()
+    newItem.createdAt = now
+    newItem.updatedAt = now
+
+    return this.itemRepository.save(newItem)
+  }
+
+  private itemShouldBeSaved(itemHash: ItemHash, existingItem?: Item): boolean {
+    if (!existingItem) {
+      return true
+    }
+
+    const incomingUpdatedAtTimestamp = itemHash.updated_at ?
+      dayjs.utc(itemHash.updated_at).valueOf() * 1000 :
+      this.timer.getTimestampInMicroseconds()
```

Comment:
Hmm..if basing on the [Ruby implementation](https://github.com/standardnotes/syncing-server/blob/develop/lib/sync_engine/2019_05_20/sync_manager.rb#L52), then we want to default to 1970 epoch datetime if updated_at is missing, and not the present time. This will ensure a conflict is properly created, which is what we want if updated_at is missing.

Reply:
Will fix 👍